lib/gui/components/modal/services/modal.js

```diff
@@ -55,23 +59,33 @@ module.exports = function($uibModal, $q) {
     return {
       close: modal.close,
       result: $q((resolve, reject) => {
-        modal.result
-          .then(resolve)
-          .catch((error) => {
+        modal.result.then((value) => {
+          AnalyticsService.logEvent('Modal accepted', value);
+
+          resolve(value);
+        }).catch((error) => {
+
+          // For some annoying reason, UI Bootstrap Modal rejects
+          // the result reason if the user clicks on the backdrop
+          // (e.g: the area surrounding the modal).
+          if (error === 'escape key press' || error === 'backdrop click') {
+            AnalyticsService.logEvent('Modal rejected', {
+              method: error
+            });
 
             // Bootstrap doesn't 'resolve' these but cancels the dialog;
-            // therefore call 'resolve' here applied to 'false'.
-            if (error === 'escape key press' || error === 'backdrop click') {
-              resolve();
+            // therefore call 'resolve' here.
+            resolve();
+
+          } else {
+            AnalyticsService.logEvent('Modal rejected', {
+              value: error
```

Comment:
Hmmm, perhaps it'd be worth copy'n'pasting the list from https://github.com/resin-io/etcher/issues/1100 onto a new page on the [wiki](https://github.com/resin-io/etcher/wiki) and then for each event that we log, make a note of what string we use to log the event, and any extra data that gets logged along with the event? I think being able to see them all at-a-glance in one place might be really useful?